[adwinang] iP

adwinang opened the pull request:
# Zephyr

> The performance of complex programs is typically optimized at the edge of chaos, just before program behaviour will become unrecognizably turbulent. - Sidney Dekker

Zephyr is simply your archaic Task Helper
- text-based
- easy to learn
- ~SUPER~ FAST to use

All you need to do is,

1. download it from [here](https://github.com/adwinang/ip)
2. go to release and download latest release
3. run
4. add your tasks
5. done

And it is **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders
- [ ] Auto Completion based on old completed tasks

If you are a Java Programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

adwinang committed:
feat: Implement Zephyr Greet and Exit

Duke file is renamed to Zephyr to follow the new name for the chatbot "Zephyr"

Zephyr simply greets and exit immediately for now

adwinang committed:
feat: Add echoing of user input and exit condition

User can exit the system by typing case insensitive "bye"
Change to usage of Archaic pronouns by Zephyr

adwinang committed:
feat: Add Task Tracker and Listing

Case insensitive keyword "list" is used to list out all the task

adwinang committed:
feat: Create Task.java for task encapsulation

Task class is incomplete, but contains enough information for a task

adwinang committed:
feat: Implement FastScanner class

FastScanner class is an abstraction on top of Java io standard library which provides useful methods for parsing user inputs in a fast, efficient and simplified way.

adwinang committed:
feat: Add todo completion tracker

Additional methods are added to Task.java to manage the task completion

adwinang committed:
feat: Add Task Subclasses

Deadline, Event and Todo classes  were added which are subclass of Task

adwinang committed:
chore: Add spacing to Event toString

adwinang committed:
feat: Add additional useful functions to FastScanner

nextUntil reads the user input till a specified delimiter. The delimiter is ignored, like String.split()

remainingLine reads the remaining string in a lline, useful to handle the cursor like functionalities of FastScanner functions

adwinang committed:
feat: Add Todo, Event and Deadline to Chatbot

adwinang committed:
fix: Default behaviour now works with full line

adwinang committed:
fix: Add missing break

adwinang committed:
test: Add test for chatbot

Used existing template and modified it for Zephyr

adwinang committed:
feat(FastScanner): Adjust nextUntil to have a not found condition

adwinang committed:
feat(ZephyrException): Add ZephyrException for better handling

adwinang committed:
feat(Zephyr.java): Add Error handling for wrong inputs

If user input is an empty string, it will also trigger an error

adwinang committed:
test: Update test for exception handling

adwinang committed:
feat: Add function to delete tasks

Changed fixed array to arraylist to allow for dynamic allocation and deallocation of array items

adwinang committed:
refactor: Change use of strings to enum for switch case

adwinang committed:
docs: Update README.md

Simplify the README.md by removing project template information

adwinang committed:
refactor: Rename variable to constant format

adwinang committed:
Revert "docs: Update README.md"

This reverts commit c0caf11cd4a6452adf2a7ea6842c880a7c1a98b1.

adwinang committed:
feat: Implement toMarkdownString for Tasks

toMarkdownString is enforced in subclasses using Task as an abstract class
toMarkdownStringInternal is used by subclasses for ease of shaping the string output

adwinang committed:
feat: Add storing and reading of save file

Additional methods were added Task subclassses to parse the partial string which ignores the markdown syntax "- [ ] {letter}:"

adwinang committed:
chore: Update gitignore to ignore save file

adwinang committed:
Merge branch 'branch-Level-7'

adwinang committed:
chore: Align method keyword

adwinang committed:
feat: Use LocalDate instead of String

For Tasks like Deadline and Event, LocalDate is used to store the details of the date instead of string

adwinang committed:
chore: Rename taskList to taskArray

This is to avoid confusion of the taskList being a list instead of an array

adwinang committed:
feat: Add command to list task with days range

adwinang committed:
chore: Update output

itsapaulblem commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,50 @@
-|____/ \__,_|_|\_\___|
+____________________________________________________________
+ Hello! I'm Zephyr
+ What can I do for thou?
```
Spelling error? Should be though?

VikramGoyal23 replied:
"Thou" is Shakespeare-era English for "you"

itsapaulblem replied:
Ah I see

itsapaulblem commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,50 @@
+     * @param partialString The markdown string after the '- [ ] D: ' part
+     * @return Returns Deadline or null if the string is not a valid Deadline
+     */
+    public static Deadline parseString(String partialString){
```
perhaps the method name should be called stringParser? as verbs are used for methods/functions

itsapaulblem reviewed (commented):
LGTM. Just nits. Keep up the good work!

adwinang committed:
refactor: Rename FastScanner to UI

adwinang committed:
feat: Create TaskList class

adwinang committed:
feat: Add Storage class

adwinang committed:
refactor: Change back to FastScanner

Remove FastScanner logic from Ui class

adwinang committed:
feat: Add saveFile function

adwinang committed:
refactor: Rename class with Task

adwinang committed:
feat: Add CommandTypes Enum

adwinang committed:
feat: Add command classes

adwinang committed:
feat: Add more functions in Ui class

Refactored print to println

adwinang committed:
fix: Remove accidental parameter

adwinang committed:
feat: Add parser to parse commands

adwinang committed:
doc: Document isValidCommand correctly

adwinang committed:
refactor: Change Task to AbstractTask

adwinang committed:
fix: Add missing refactor

adwinang committed:
fix: Add missing output for commands

adwinang committed:
feat: Implemented Zephyr with updated classes

adwinang committed:
refactor: Move all files under Zephyr folder

adwinang committed:
refactor: Enforce immutability by adding final

adwinang committed:
refactor: Move all files to corresponding packages

adwinang committed:
Merge branch 'add-gradle-support'

adwinang committed:
fix: Fix missing task type checking

Add getTaskType to Task class to be able check the string name of the task class

adwinang committed:
fix: Fix incorrect imports and missing public

adwinang committed:
feat: Add getter for description

adwinang committed:
fix: Add missing markAsUndone

adwinang committed:
test: Add DeadlineTaskTest

adwinang committed:
fix: Correct return type for To getter

adwinang committed:
test: Add EventTaskTest

adwinang committed:
docs(AbstractCommand): Add Javadoc

adwinang committed:
docs(AbstractCommand.java): Adjust comment with @code

adwinang committed:
docs(ByeCommand): Add Javadoc

adwinang committed:
docs(DeadlineCommand.java): Add Javadocs

adwinang committed:
docs(DeleteCommand.java): Add Javadocs

adwinang committed:
docs(EventCommand.java): Add Javadocs

adwinang committed:
docs(ListCommand.java): Add Javadocs

adwinang committed:
docs(MarkCommand.java): Add Javadocs

adwinang committed:
docs: Add remaining Javadocs for remaining Commands

adwinang committed:
docs: Add Javadocs for all controllers

adwinang committed:
docs: Adjust Javadocs to have no html tags

adwinang committed:
docs: Add Javadocs for all remaining files

adwinang committed:
fix: Change wrong package name for CommandTypes

adwinang committed:
feat: Add find command feature

adwinang committed:
Merge branch 'branch-A-CodingStandard'

adwinang committed:
Merge branch 'branch-Level-9'

adwinang committed:
fix: Add missing command in command list

adwinang committed:
docs(Ui.java): Fix styling

adwinang committed:
refactor: Move all files under java instead of under zephyr

adwinang committed:
fix: Change package name

adwinang committed:
refactor: Use getter for words and arguments

adwinang committed:
refactor: Fix all CheckStyle errors

adwinang committed:
feat: Add javaFx library

adwinang committed:
feat: Interation 1

Following the guide here
https://se-education.org/guides/tutorials/javaFxPart3.html

adwinang committed:
feat: Add to queue instead of printing to console in Ui

adwinang committed:
feat: Use fxml for ui implementation

adwinang committed:
fix: Add missing new line to addBreak function

adwinang committed:
feat: Add background to ui

adwinang committed:
feat: Add one function to implement varargs

adwinang committed:
refactor: Change from print to show

adwinang committed:
docs: Fix minor documentation

This is to avoid triggering the Check Style regex

adwinang committed:
docs: Fix CheckStyle regex error

adwinang committed:
feat(EventCommand): Add assert

adwinang committed:
fix: Remove unnecessary parameters for parser

adwinang committed:
Merge pull request #2 from adwinang/branch-A-Assertions

Branch A Assertions

adwinang committed:
Merge pull request #3 from adwinang/branch-A-CodeQuality

fix: Remove unnecessary parameters for parser

adwinang committed:
feat: Add tagging functionality

adwinang committed:
Merge pull request #4 from adwinang/branch-C-Tagging

feat: Add tagging functionality

adwinang committed:
feat: Make some adjustments to ui

adwinang committed:
Merge pull request #5 from adwinang/branch-A-BetterGui

feat: Make some adjustments to ui

adwinang committed:
refactor: Move some addQueue to Ui function

adwinang committed:
chore: Remove Break lines for content

adwinang committed:
Merge pull request #6 from adwinang/branch-A-Personality

Branch a personality

adwinang committed:
feat: Add check for at date and end date

adwinang committed:
Merge pull request #7 from adwinang/branch-A-MoreErrorHandling

feat: Add check for at date and end date

adwinang committed:
test: Fix test errors

adwinang committed:
test: Add TodoTaskTest

adwinang committed:
Merge pull request #8 from adwinang/branch-A-MoreTesting

Branch a more testing

adwinang committed:
refactor: Change name of duke to zephyr

adwinang committed:
docs(README.md): Change Duke to Zephyr

adwinang committed:
fix: Add logic to catch Zephyr exceptions in window

adwinang committed:
refactor: Move file saving logic to bye command

adwinang committed:
fix: Remove break lines

adwinang committed:
fix: Fix the loading of file

adwinang committed:
refactor: Change format of tags

adwinang committed:
fix: Standardise DateTime formatting

adwinang committed:
fix: Add parsing of tags from storage

adwinang committed:
feat: Add welcome message to app

adwinang committed:
feat: Add title to app window

adwinang committed:
refactor: Change /at to /to

adwinang committed:
docs: Add user guide and support image

adwinang committed:
Merge pull request #9 from adwinang/branch-A-UserGuide

docs: Add user guide and support image

adwinang committed:
docs: Update README.md

adwinang committed:
fix: Change javaFX version in fxml

adwinang committed:
fix: Corrected test files for previous changes

adwinang committed:
fix: Add missing launcher class

Missing launcher class was causing the jar file to not run on other OS environments
